exceptions/pulp/exceptions/base.py

```diff
@@ -1,93 +1,44 @@
-        self.error_data = {}
-
-        # child exceptions are those that are wrapped within this exception, validation errors
-        # for example would have one overall validation error and then a separate sub error
-        # for each validation that failed
-
-        self.child_exceptions = []
-
-    def add_child_exception(self, exception):
-        self.child_exceptions.append(exception)
-
-    def to_dict(self):
+    def __init__(self, error_code):
         """
-        The to_dict method is used to provide a standardized dictionary
-        of the exception information for usage storing to the database
-        or converting to json to send back via an API call
+        :param error_code: unique error code
+        :type error_code: str
         """
-        result = {
-            'code': self.error_code.code,
-            'description': str(self),
-            'data': self.error_data,
-            'sub_errors': []
-        }
-        for error in self.child_exceptions:
-            if isinstance(error, PulpException):
-                result['sub_errors'].append(error.to_dict())
-            else:
-                result['sub_errors'].append({'code': 'PLP0000',
-                                             'description': str(error),
-                                             'data': {},
-                                             'sub_errors': []})
-        return result
-
-    def __str__(self):
-        class_name = self.__class__.__name__
-        msg = _('Pulp exception occurred: %(c)s') % {'c': class_name}
-        if self.args and isinstance(self.args[0], basestring):
-            msg = self.args[0]
-        return msg.encode('utf-8')
-
-    def data_dict(self):
-        return {'args': self.args}
-
-
-class PulpCodedException(PulpException):
-    """
-    Base class for exceptions that put the error_code and data as init arguments
-    """
-    def __init__(self, error_code=error_codes.PLP0001, **kwargs):
-        super(PulpCodedException, self).__init__()
+        if not isinstance(error_code, basestring):
+            raise RuntimeError("Error code must be an instance of basestring.")
         self.error_code = error_code
-        if kwargs:
-            self.error_data = kwargs
-        # Validate that the coded exception was raised with all the error_data fields that
-        # are required
-        for key in self.error_code.required_fields:
-            if key not in self.error_data:
-                raise PulpCodedException(error_codes.PLP0008, code=self.error_code.code,
-                                         field=key)
 
     def __str__(self):
-        msg = self.error_code.message % self.error_data
-        return msg.encode('utf-8')
+        """
+        Returns the string representation of the exception.
+
+        Each concrete class that inherits from :class: `pulp.server.exception.PulpException` is
+        expected to implement it's own __str__() method.
+        """
+        pass
```

Comment:
This should probably be `NotImplementedError` so that subclasses are required to implement it.